src/language/JSUtils.js

```diff
@@ -68,19 +57,99 @@ define(function (require, exports, module) {
     function _findAllFunctionsInText(text) {
         var results = {},
             functionName,
+            resultNode,
+            memberPrefix,
             match;
-
+   
         PerfUtils.markStart(PerfUtils.JSUTILS_REGEXP);
-
-        while ((match = _functionRegExp.exec(text)) !== null) {
-            functionName = (match[2] || match[5]).trim();
-
+        
+        var AST = AcornLoose.parse_dammit(text, {locations: true});
```

Comment:
I think this work by checking indentation of the code.
So you need
1) use always first Acorn.parse and if it fails use AcornLoose.parse_dammit
2) add some tests of code not indented correctly (and / or without indentation at all)